examples/postgresql/README.md

```diff
@@ -0,0 +1,109 @@
+# User authentication in Postgresql using SVIDs with DNS names
+
+This is an example of how **spiffe-helper** can be used to authenticate users to a **PostgreSQL** database using x509 SVIDs.
+
+Postgres provides a feature that allows users to authenticate using [certificate authentication](https://www.postgresql.org/docs/9.5/auth-methods.html#AUTH-CERT). The server validates that the CN (Common Name) attribute of the certificate presented by the client matches the database user name or a [mapped](https://www.postgresql.org/docs/9.5/auth-username-maps.html) value.
+
+This guide sets up the authentication configuration for the username: `postgres-user`.
+
+
+## Guide
+
+### Considerations
+The following assumptions are made:
++ At least one SPIRE server and one agent are deployed with trust domain `example.org`.
+
++ Postgres 12 is used in this guide but each step should be easily applicable to other versions as well.
+
+### 1. Install PosgreSQL
+Install [PosgreSQL](https://www.postgresql.org/docs/12/tutorial-install.html) and make sure the service is up running.
+```
+systemctl status postgresql@12-main
+```
+
+### 2. Create the user
+Create the `postgres-user` using the [provided script](create_user.sql).
+```sql
+sudo -u postgres psql -c create_user.sql
+```
+
+### 3. Configure PosgreSQL SSL settings
+Make PostgreSQL server to use the certificates and key provided by SPIRE by setting the following SSL configurables in the PostgreSQL configuration file (`postgresql.conf`).
+```bash
+ssl = on
+ssl_cert_file = '/opt/spire/certs/postgresql/svid.pem'
+ssl_key_file = '/opt/spire/certs/postgresql/svid.key'
+ssl_ca_file = '/opt/spire/certs/postgresql/svid_bundle.pem'
+```
+
+Create the folder using the `postgres` user:
+```
+sudo -u postgres mkdir -p /opt/spire/certs/postgresql
+```
+
+### 4. Configure pg_hba.conf
+Create the following rules in the `pg_hba.conf` file:
+```
+# TYPE      DATABASE        USER            ADDRESS                 METHOD
+
+# Allows postgres user to connect using unix domain sockets (not required)
+local       all             postgres                                trust
+
+# Reject all nossl calls (not required)
+hostnossl   all             all             0.0.0.0/0               reject
+
+# Validate provided certificates
+hostssl     all             all             0.0.0.0/0               cert clientcert=1
+```
+
+### 5. Start SPIRE server
+Start the SPIRE server:
+```bash
+./spire-server run
+```
+
+### 6. Start SPIRE agent
+Start the SPIRE agent:
+```bash
+./spire-server bundle show > conf/agent/dummy_root_ca.crt
```

Comment:
what happens in case of upstream bundle is configured? (if user want to have a long live certificate as CA?)